auth/auth.go

```diff
@@ -38,5 +42,58 @@ func main() {
 }
 
 func authCreateHandler(w http.ResponseWriter, r *http.Request) {
-	json.NewEncoder(w).Encode(struct{}{})
+	var req authDAO.AuthCreateRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	_, err = valid.ValidateStruct(req)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	// Hash and salt the password before storing
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(req.Password), bcrypt.DefaultCost)
```

Comment:
Is the salt something implicit or does it need to be stored?

Reply:
The salt is handled by the beautiful people who build `bcrypt`: 

https://github.com/golang/crypto/blob/2aa609cf4a9d7d1126360de73b55b6002f9e052a/bcrypt/bcrypt.go#L145